### PR TITLE
Review of ZCE extensions - Sail Code V 0.5

### PR DESCRIPTION
**Pull request by junambi**
Split the big file into three smaller files as follows.

model/riscv_insts_zcb.sail
model/riscv_insts_zcmp.sail
model/riscv_insts_zcmt.sail

Let us start the review with the first file, then move on to the other ones.

Thank you.

### CODE REVIEW

File: model/riscv_insts_zcb.sail [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+  let rsext = X(rs) + immext;
+  X(rs) = rsext;
+
+   match execute(LOAD(imm, rs, rd, false, WORD, false, false)) {
+     RETIRE_SUCCESS => {
+                         X(rd) = EXTZ(X(rd)[7..0]);
+                         RETIRE_SUCCESS
+                       },
+     RETIRE_FAIL => RETIRE_FAIL,
+   }
```
Comment (jrtc27) [inline model/riscv_insts_zcb.sail:100]: These are still completely wrong. Stop wasting our time.

**Comment by abukharmeh** [comment]
Hi, thanks for splitting the file. Is this really version 0.5 of the spec ? 
Why didnt you fix the load/ store size and added an enable switch to all the encdec statements like requested in the earlier PR ?  

I realise you said it didnt work when you had the correct size, can you at least let us know what was the error reported by SAIL ? 
Similarly, can you let us know what was the error reported when you removed the match statements I called redundant earlier  ? 

Ideally I would like to fill the following table with complete (For Zcb) but I didn't try to really fill all my comments as the issues I raised in earlier PR are still not addressed ?

| Instruction | Encoding | Functionality                                                     | Style | Comments |
|-------------|----------|-------------------------------------------------------------------|-------|----------|
| c.lbu       |          | Wrong load size ?                                                 |       |          |
| c.lhu       |          | Wrong load size ?                                                 |       |          |
| c.lh        |          | Wrong load size ?                                                 |       |          |
| c.sb        |          | Wrong store Size ?                                                |       |          |
| c.sh        |          | Wrong store size ?                                                |       |          |
| c.zext.b    |          |                                                                   |       |          |
| c.sext.b    |          |                                                                   |       |          |
| c.zext.h    |          |                                                                   |       |          |
| c.sext.h    |          |                                                                   |       |          |
| c.zext.w    |          |                                                                   |       |          |
| c.mul       |          | Could be mapped to mul similar to other compressed instructions ! |       |          |
| c.not       |          |                                                                   |       |          |

**Comment by ptomsich** [comment]
> Hi, thanks for splitting the file. Is this really version 0.5 of the spec ? Why didnt you fix the load/ store size and added an enable switch to all the encdec statements like requested in the earlier PR ?
> 
> I realise you said it didnt work when you had the correct size, can you at least let us know what was the error reported by SAIL ? Similarly, can you let us know what was the error reported when you removed the match statements I called redundant earlier ?

@abukharmeh If you reread the execute-functions, you will see that not just the load-size is incorrect, but also a spurious pre-load update of `rs` occurs (i.e., the immediate is added to `rs` and written back — and then added in the execution of the LOAD again).

> Ideally I would like to fill the following table with complete (For Zcb) but I didn't try to really fill all my comments as the issues I raised in earlier PR are still not addressed ?
> 
> Instruction	Encoding	Functionality	Style	Comments
> c.lbu		Wrong load size ?		
> c.lhu		Wrong load size ?		
> c.lh		Wrong load size ?		
> c.sb		Wrong store Size ?		
> c.sh		Wrong store size ?		
> c.zext.b				
> c.sext.b				
> c.zext.h				
> c.sext.h				
> c.zext.w				
> c.mul		Could be mapped to mul similar to other compressed instructions !		
> c.not

**Comment by abukharmeh** [comment]
@ptomsich yes indeed, forgot to mention this again here, my comment about rsext in the old PR !

**Comment by junambi** [comment]
I think i will close this PR as well, work on the changes and then open a new PR, sorry for the confusion created.

Thank you for the comments and inputs though.

**Pull request closed without merging**